src/fragments/lib/storage/flutter/copy.mdx

```diff
@@ -11,12 +7,12 @@ Future<void> copyGuestFileToPrivate({
 }) async {
   try {
     final result = await Amplify.Storage.copy(
-      source: S3ItemWithAccessLevel(
-        storageItem: S3Item(key: sourceKey),
+      source: StorageItemWithAccessLevel(
+        storageItem: StorageItem(key: sourceKey),
         accessLevel: StorageAccessLevel.guest,
       ),
-      destination: S3ItemWithAccessLevel(
-        storageItem: S3Item(key: destinationKey),
+      destination: StorageItemWithAccessLevel(
+        storageItem: StorageItem(key: destinationKey),
```

Comment:
Just realized it's a bit weird to take a storage item for the destination instead of just a key. Even the source too - maybe these should be just keys?

Reply:
key is the only required field for instantiating a `StorageItem`. @HuiSF wdyt?

Reply:
This pattern was following the combination of amplify-js (has copy API) and amplify-android (has StorageItem), plus feature extensibility. 

One example of advantages of using `StorageItem` instead of a string key is that the possibility of supporting metadata updating during copying (this is not planned for initial v1).

```dart
  final result = await Amplify.Storage.copy(
    source: const StorageItemWithAccessLevel(
      storageItem: StorageItem(key: 'old_key'),
      accessLevel: StorageAccessLevel.guest,
    ),
    destination: const StorageItemWithAccessLevel(
      storageItem: StorageItem(
        key: 'new_key',
        metadata: {'new_field': 'new value'},
      ),
      accessLevel: StorageAccessLevel.private,
    ),
  ).result;
```

Thoughts?

Reply:
thanks @HuiSF, it makes sense to use StorageItem.

Reply:
Cool! I didn't realize you could update metadata this way.